src/matching/quoteMatcher.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * QuoteMatcher matches quoted strings, respecting escaped quotes (\") and frieds
+ */
+export class QuoteMatcher {
+  static escapeChar = "\\";
+
+  static findOpening(char: string, corpus: string, start: number): number {
+    // first, search backwards to see if we could be inside a quote
```

Comment:
This comment is somewhat inconsistent with the code below. You are going to check whether we are inside a quote, the code is seaching for a quote backwards. But what if there are two paired quotes on the left side? Finding the closing one doesn't mean we are inside a quote, right?

Did I miss something or is it a Vim design?